src/main/java/org/killbill/billing/plugin/adyen/client/payment/service/AdyenPaymentRequestSender.java

```
@@ -29,49 +35,156 @@
+                return new UnSuccessfulAdyenCall<T>(RESPONSE_NOT_RECEIVED, e.getMessage());
+            } else if (rootCause.getMessage().contains("Unexpected end of file from server")) {
+                return new UnSuccessfulAdyenCall<T>(RESPONSE_INVALID, rootCause.getMessage());
+            }
+        } else if (rootCause instanceof SocketException) {
+            if (rootCause.getMessage().contains("Unexpected end of file from server")) {
+                return new UnSuccessfulAdyenCall<T>(RESPONSE_INVALID, e.getMessage());
+            }
+        } else if (rootCause instanceof HTTPException) {
+            // connection timeout
+            if (((HTTPException) rootCause).getResponseCode() == SERVICE_UNAVAILABLE) {
+                return new UnSuccessfulAdyenCall<T>(REQUEST_NOT_SEND, rootCause.getMessage());
+            }
+            // e.g. different response code or strange response
+            return new UnSuccessfulAdyenCall<T>(RESPONSE_INVALID, rootCause.getMessage());
+        } else if (rootCause instanceof SOAPFaultException) {
+            return new UnSuccessfulAdyenCall<T>(RESPONSE_INVALID, e.getMessage());
+        } else if (rootCause instanceof IOException) {
+            if (rootCause.getMessage().contains("Invalid Http response")) {
+                // unparsable data as response
+                return new UnSuccessfulAdyenCall<T>(RESPONSE_INVALID, e.getMessage());
+            } else if (rootCause.getMessage().contains("Bogus chunk size")) {
+                return new UnSuccessfulAdyenCall<T>(RESPONSE_INVALID, e.getMessage());
+            }
+        } else if (rootCause instanceof WstxEOFException) {
+            // happens for example when 301 with empty body is returned...
+            return new UnSuccessfulAdyenCall<T>(RESPONSE_INVALID, e.getMessage());
+        } else if (rootCause instanceof SoapFault) {
+            return new UnSuccessfulAdyenCall<T>(RESPONSE_ABOUT_INVALID_REQUEST, e.getMessage());
```

All business errors are not reported als SoapFaults but instead as valid responses with status Refused (which is in my eyes also cleaner). SoapFault is like a IllegalArgument or NullpoinerException. So it's a programming error on our side. For example we missed to provide the psp ref. So it's nothing which depends on the user.

*But* the plugin contains also the AdyenPaymentExceptionParser which actually tried to parse business validation errors from SOAPExceptions. So either the parser never worked or i am wrong here.